### ISSUE TITLE
LLM finetuning

### ISSUE THREAD

**Issue by jeonghunyoon** [jeonghunyoon/Recommendation#6]
- Llama : https://llama.meta.com/docs/how-to-guides/fine-tuning
- Quantization : 실수형 변수(float)를 정수형 변수(int)로 변환
  - 효과
    - 모델 사이즈 축소
    - 모델 연산량 감소
    - 효율적인 하드웨어 사용
- Parameter Efficient Fine Tuning
  - 모델은 freezing & 학습가능한 tiny parameters/layers 모델에 추가 ==> 결국, 모델의 매우 적은 부분의 parameters만 학습
- Prompting이란 무엇인가?
- Prompting method의 2가지 카테고리
  - Hard prompt
  - Soft prompt
    - Prompt tuning
    - Prefix tuning
    - P-tuning

**Comment by jeonghunyoon** [comment]
## Prompt란?
- https://huggingface.co/docs/peft/main/en/task_guides/prompt_based_methods
- A **prompt** can **describe a task** or **provide an example of a task** _you want the model to learn_.
  - `PromptTuningConfig`
    - ex) `prompt_tuning_init_text="Classify if the tweet is a complaint or not:",`
      - input data에 `Tweet text`, `Label` word가 포함되어 있음
- **Soft prompting** methods **add learnable parameters to the input embeddings** _that can be optimized for a specific task_ while keeping the pretrained model’s parameters frozen.
